src/main/java/org/example/CalculatorFactory.java

```diff
@@ -0,0 +1,14 @@
+package org.example;
+
+public class CalculatorFactory {
+
+    public Calculator getInstance(String type){
+        if(type.equals("Basic"))
```

Comment:
Using switch here would be better than if-else chains as there could be any number of additions to the factory in the future.